tests/func/test_import.py

```diff
@@ -725,12 +725,40 @@ def test_import_invalid_configs(tmp_dir, scm, dvc, erepo_dir):
         )
 
 
-def test_import_no_hash(tmp_dir, scm, dvc, erepo_dir, mocker):
+@pytest.mark.parametrize(
+    "files",
+    [
+        {"foo": "foo"},
+        {
+            "dir": {
+                "bar": "bar",
+                "subdir": {"lorem": "ipsum", "nested": {"lorem": "lorem"}},
+            }
+        },
+    ],
+)
+def test_import_no_hash(tmp_dir, scm, dvc, erepo_dir, mocker, files):
     with erepo_dir.chdir():
-        erepo_dir.dvc_gen("foo", "foo content", commit="create foo")
+        erepo_dir.dvc_gen(files, commit="create foo")
 
     spy = mocker.spy(hash, "file_md5")
-    stage = dvc.imp(os.fspath(erepo_dir), "foo", "foo_imported")
-    assert spy.call_count == 1
-    for call in spy.call_args_list:
-        assert stage.outs[0].fs_path != call.args[0]
+    index_info_spy = mocker.spy(DataIndex, "info")
+    name = next(iter(files))
+    dvc.imp(os.fspath(erepo_dir), name, "out")
+
+    local_hashes = [
+        call.args[0] for call in spy.call_args_list if call.args[1].protocol == "local"
+    ]
+    assert not local_hashes
+
+    expected_info_calls = {(name,)}
+    if isinstance(files[name], dict):
+        dirs = {
+            (name, *d.relative_to(tmp_dir / "out").parts)
+            for d in (tmp_dir / "out").rglob("*")
+            if d.is_dir()
+        }
+        expected_info_calls.update(dirs)
+    assert {
+        call.args[1] for call in index_info_spy.call_args_list
+    } == expected_info_calls
```

Comment:
asserting that `Index.info()` calls only happen for directories during `fs.walk()`.

Reply:
Not that strong an opinion, but would it be better to have a separate test for files and dirs? Maybe we need to leave some comments? This isn't the easiest test to understand on its own.

Reply:
I have simplified the tests. PTAL.